extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/ColumnChunkPageStore.java

```diff
@@ -48,25 +53,60 @@ private CreatorResult(@NotNull final ColumnChunkPageStore<ATTR> pageStore,
         }
     }
 
+    private static boolean canUseOffsetIndexBasedPageStore(@NotNull final ColumnChunkReader columnChunkReader,
+            @NotNull final ColumnDefinition<?> columnDefinition) {
+        if (columnChunkReader.getOffsetIndex() == null) {
+            return false;
+        }
+        final String version = columnChunkReader.getVersion();
+        if (version == null) {
+            // Parquet file not written by deephaven, can use offset index
+            return true;
+        }
+        // For vector and array column types, versions before 0.31.0 had a bug in offset index calculation, fixed as
+        // part of deephaven-core#4844
+        final Class<?> columnType = columnDefinition.getDataType();
+        if (columnType.isArray() || Vector.class.isAssignableFrom(columnType)) {
+            return satisfiesMinimumVersionRequirements(version);
+        }
+        return true;
+    }
+
+    private static final Pattern VERSION_PATTERN = Pattern.compile("(\\d+)\\.(\\d+)\\.(\\d+)");
+
+    /**
+     * Check if the version is greater than 0.31.0
+     */
+    @VisibleForTesting
+    public static boolean satisfiesMinimumVersionRequirements(@NotNull final String version) {
+        final Matcher matcher = VERSION_PATTERN.matcher(version);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException("Malformed version:" + version);
+        }
+        final int major = Integer.parseInt(matcher.group(1));
+        final int minor = Integer.parseInt(matcher.group(2));
+        return major > 0 || major == 0 && minor >= 31;
+    }
+
     public static <ATTR extends Any> CreatorResult<ATTR> create(
             @NotNull final PageCache<ATTR> pageCache,
             @NotNull final ColumnChunkReader columnChunkReader,
             final long mask,
-            @NotNull final ToPage<ATTR, ?> toPage) throws IOException {
-        final boolean fixedSizePages = columnChunkReader.getPageFixedSize() >= 1;
-        final ColumnChunkPageStore<ATTR> columnChunkPageStore = fixedSizePages
-                ? new FixedPageSizeColumnChunkPageStore<>(pageCache, columnChunkReader, mask, toPage)
+            @NotNull final ToPage<ATTR, ?> toPage,
+            @NotNull final ColumnDefinition<?> columnDefinition) throws IOException {
+        final boolean canUseOffsetIndex = canUseOffsetIndexBasedPageStore(columnChunkReader, columnDefinition);
```

Comment:
It's unfortunate we have to weave the logic like this - the pervasive passing of `version` now is less than ideal IMO. Maybe we want to have the implementations have access to the full `TableInfo` instead?

Reply:
I tried passing `TableInfo` instead of `version` to `ParquetFileReader` class but that would require adding a dependency of `Deephaven_Community_Core.extensions-parquet-base.main` on `Deephaven_Community_Core.extensions-parquet-table.main` leading to circular dependency.